onnxruntime/core/framework/session_state.cc

```diff
@@ -856,7 +856,8 @@ void SessionState::UpdateToBeExecutedNodes(const std::vector<int>& fetch_mlvalue
   // Get the nodes generating the fetches.
   std::vector<const Node*> nodes;
   nodes.reserve(fetch_mlvalue_idxs.size());
-  std::unordered_set<NodeIndex> reachable_nodes;
+  InlinedHashSet<NodeIndex> reachable_nodes;
+  reachable_nodes.reserve(fetch_mlvalue_idxs.size());
```

Comment:
>fetch_mlvalue_idxs.size() [](http://example.com/codeflow?start=26&length=25)

I think this could be larger too, potentially the number of nodes in graph_ if everything is reachable